File: 백준/Silver/9417. 최대 GCD/최대 GCD.py
n = int(input())

def get_gcd(a, b):
    while a % b != 0:
        a, b = b, a % b
    return min(a, b)

for _ in range(n): #100개
    arr = list(map(int, input().split())) #100개
    ans = 0
    for i in range(len(arr) - 1):
        for j in range(i + 1, len(arr)):
            ans = max(ans ,get_gcd(max(arr[i],arr[j]),min(arr[i],arr[j])))
    print(ans)

File: 백준/Silver/9417. 최대 GCD/최대 GCD.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st ;
        StringBuilder sb = new StringBuilder();
        int n = Integer.parseInt(br.readLine());
        ArrayList<Integer> arr = new ArrayList<>();

        for (int i = 0; i < n; i++) {
            arr.clear();
            st = new StringTokenizer(br.readLine());
            int ans = 0;
            while(st.hasMoreTokens()) {
                arr.add(Integer.parseInt(st.nextToken()));
            }

            for (int j = 0; j < arr.size() - 1; j++) {
                for (int k = j + 1; k < arr.size(); k++) {

                    ans = Math.max(ans, gcd(Math.max(arr.get(j), arr.get(k)),Math.min(arr.get(j), arr.get(k))));
                }
            }
            sb.append(ans + "\n");
        }
        System.out.println(sb);

    }

    public static int gcd(int a, int b){
        while(a % b != 0){
            int tmp = a;
            a = b;
            b = tmp % b;
        }
        return b;
    }
}
